File: update_pbxproj.rb
require 'xcodeproj'

project_path = '/Users/okasha/clicky/leanring-buddy.xcodeproj'
project = Xcodeproj::Project.open(project_path)
target = project.targets.first

group = project.main_group.find_subpath('leanring-buddy', false)

# Files deleted from disk — remove from Xcode project
files_to_remove = [
  'LocalVisionProcessor.swift',
  'ElementLocationDetector.swift'
]

# New files added to disk — add to Xcode project
files_to_add = [
  'AgentToolDefinition.swift',
  'AgentToolExecutor.swift',
  'AgentLoop.swift'
]

# Remove deleted files
files_to_remove.each do |file_name|
  file_ref = group.files.find { |f| f.path == file_name || f.name == file_name }
  if file_ref
    puts "Removing #{file_name} from project"
    target.source_build_phase.remove_file_reference(file_ref)
    file_ref.remove_from_project
  else
    puts "#{file_name} not found in project (already removed?)"
  end
end

# Add new files
files_to_add.each do |file_name|
  # Skip if already in the project
  if group.files.any? { |f| f.path == file_name || f.name == file_name }
    puts "#{file_name} already in project — skipping"
    next
  end

  file_path = File.join(group.real_path, file_name)
  unless File.exist?(file_path)
    puts "WARNING: #{file_name} not found on disk at #{file_path}"
    next
  end

  puts "Adding #{file_name} to project"
  file_ref = group.new_reference(file_path)
  target.source_build_phase.add_file_reference(file_ref)
end

project.save
puts "Project saved successfully"

File: update_pbxproj.py
import os
from pbxproj import XcodeProject

project_path = '/Users/okasha/clicky/leanring-buddy.xcodeproj/project.pbxproj'
project = XcodeProject.load(project_path)

files_to_remove = [
    'AssemblyAIStreamingTranscriptionProvider.swift',
    'ClaudeAPI.swift',
    'ClickyAnalytics.swift',
    'ElevenLabsTTSClient.swift',
    'OpenAIAPI.swift',
    'OpenAIAudioTranscriptionProvider.swift'
]

files_to_add = [
    'leanring-buddy/LocalTTSClient.swift',
    'leanring-buddy/LocalVisionProcessor.swift',
    'leanring-buddy/Mem0Client.swift',
    'leanring-buddy/OllamaAPI.swift',
    'leanring-buddy/OllamaModelMemoryManager.swift'
]

for file in files_to_remove:
    # We find files by their name
    file_refs = project.get_files_by_name(file)
    if file_refs:
        for file_ref in file_refs:
            print(f"Removing {file}")
            project.remove_file_by_id(file_ref.get_id())
    else:
        print(f"File {file} not found in project")

for file_path in files_to_add:
    print(f"Adding {file_path}")
    project.add_file(file_path, parent=project.get_or_create_group('leanring-buddy'), force=False)

project.save()
print("Project saved successfully.")
